File: cpp/chapter1/printEvenNumbers.cpp
#include<iostream>
using namespace std;

void printEvenNumber1(int n){
     int i=0;
    while(i <= n){
        if(i%2==0)
        cout<<i<<endl;
        i=i+1;
    }
}

void printEvenNumber2(int n){
     int i=0;
    while(i <= n){
        cout<<i<<endl;
        i=i+2;
    }
}

int main(){
    //this program compares the efficience of two similar algorithms.
    printEvenNumber1(30);
    printEvenNumber2(30);

   
    return 0;
}

File: python/chapter1/printEvenNumbers.py

def printEvenNumber1(limit):
    numbers = 0
    while numbers <= limit:
        if numbers%2 ==0:
            print(numbers)
        numbers=numbers+1

def printEvenNumber2(limit):
    numbers = 0
    while numbers <= limit:
        print(numbers)
        numbers=numbers+2

def main():
    limit = 100
    printEvenNumber1(limit)
    printEvenNumber2(limit)

main()